src/Compilers/CSharp/Portable/Binder/Semantics/Conversions/Conversions.cs

```diff
@@ -312,8 +312,7 @@ public override Conversion GetStackAllocConversion(BoundStackAllocArrayCreation
 
                         if (spanConversion.Exists)
                         {
-                            // Report errors if Span ctor is missing, or using an older C# version
-                            Binder.CheckFeatureAvailability(sourceExpression.Syntax, MessageID.IDS_FeatureRefStructs, ref useSiteDiagnostics);
+                            // Report errors if Span ctor is missing
                             Binder.GetWellKnownTypeMember(_binder.Compilation, WellKnownMember.System_Span_T__ctor, out DiagnosticInfo memberDiagnosticInfo);
```

Comment:
>Binder.GetWellKnownTypeMember(_binder.Compilation, WellKnownMember.System_Span_T__ctor, out DiagnosticInfo memberDiagnosticInfo); [](start = 28, length = 129)

I don't think we should be performing this check here either. Whether the constructor is good has no effect on the result of conversion classification, we still return the same Conversion value. Instead, the use-site diagnostics for the constructor should be reported where we actually try to use it. I see two other places where we look for the same symbol:
```
  src\Compilers\CSharp\Portable\Binder\Binder_Expressions.cs(3128):                GetWellKnownTypeMember(Compilation, WellKnownMember.System_Span_T__ctor, diagnostics, syntax: node);
  src\Compilers\CSharp\Portable\Lowering\LocalRewriter\LocalRewriter_StackAlloc.cs(42):                var spanCtor = (MethodSymbol)_compilation.GetWellKnownTypeMember(WellKnownMember.System_Span_T__ctor).SymbolAsMember(spanType);
```

The first one also doesn't use the symbol, simply checks its "goodness". I think we should remove that check as well, and the goodness should only be checked in LocalRewriter, which does not do this at the moment. See how LocalRewriter.TryGetWellKnownTypeMember helper is used elsewhere in the rewriter.